src/snowcli/output/decorators.py

```diff
@@ -25,3 +28,25 @@ def _is_list_of_results(result):
         and len(result) > 0
         and (isinstance(result[0], list) or isinstance(result[0], SnowflakeCursor))
     )
+
+
+def catch_error(
+    exception_class: Type[Exception], message: Optional[str] = None, exit_code: int = 1
+):
+    """
+    Catches a specific type of exception and exits fatally, optionally with
+    a custom message or process exit code.
+    """
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except exception_class as e:
+                print(message if message else str(e), file=stderr)
+                raise typer.Exit(code=exit_code)
+
+        return wrapper
+
+    return decorator
```

Comment:
The idea of throwing always the same error is reasonable. However, I'm not sure if we need decorator. We can probably implement this on app level. @sfc-gh-pjob wdyt?

Reply:
Currently we catch all unhandled exceptions on the app level (main typer level), we print information about unhandled exception and exit with "1" code. User also can use `--debug` option to have full traceback instead of it. To make this working the command has to be decorated with `@global_options` or `@global_options_with_connection`. If it is not decorated, then `--debug` mode is by default.

The message without `--debug` looks like the following:
```
An unexpected exception occurred. Use --debug option to see the traceback.
Exception message:
Test exception message
```

This is what we have on the app level.

---

As I can see this decorator tries to achieve sth different. 
It allows to catch errors about which you are sure that it is safe to catch and silent them.
As long as it is used to catch exceptions generated by our code, and for which we are sure we don't need tracebacks, it offers cleaner messages. 

You receive:
```
Message extracted from test exception
```
or:
```
Custom message
```
instead of:
```
An unexpected exception occurred. Use --debug option to see the traceback.
Exception message:
Message extracted from test exception
```

IMO this decorator is ok if it is used only to simplify returning errors generated by ourselves but it may cause issues with errors debugging if used to catch exceptions raised either not by us or in multiple places in our code.

---

But I'm not sure if we need such decorator and if we need to add anything on the app level.
There is `ClickException` which has a few child classes delivered by Click and next few implemented by us.
We usually use simply `raise XyzException("wrong path provided")`, where `XyzException` is extending `ClickException` or one of its children. Extending the `ClickException` allows Click to catch the Exception on the framework level, print it nicely with colours and exit with proper exit code. It all is being done before our global handling.

IMO we should extend and use Click exceptions as often as possible because it guarantees consistent formatting of our errors.

Please take a look at https://click.palletsprojects.com/en/7.x/api/#click.ClickException and at `exception.py` in our codebase. If you will want to add custom exceptions for native apps functionality, then keep it in your subpackage (as you're doing it now) and do not add them to `exception.py` because this module will be part of public API for plugins.

Reply:
This is very helpful, thanks for the feedback! I think the two big differences are:
- ability to configure exit code
- removal of the traceback

Happy to just extend `ClickException` and remove this decorator. It would be nice to have the exit code configurable but it's not a hill I'm willing to die on :D

Reply:
IMO you can customise exit_code. Take a look at `UsageError` which is provided by Click and which extends `ClickException`. Default exit_code for `ClickException` is 1 but `UsageError` overrides it and uses exit_code = 2.